# SVM with Kernel: Polynomial

In [2]:
# import the necessary packages
import time
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import classification_report,confusion_matrix

In [3]:
df = pd.read_csv('mnist.csv')       # read the csv file

df_feat = df.drop('label',axis=1)   # drop the label column

label_y = df['label']               # label column

In [5]:
x_train, x_test, y_train, y_test = train_test_split(df_feat,label_y,test_size=0.5)

model = SVC(kernel='poly',degree=2) # kernel='poly',degree=2

start = time.time()
model.fit(x_train,y_train)
end = time.time()
print("Time taken to train the model: %s min %s sec"% ((end-start)//60,(end-start)%60))
predictions = model.predict(x_test)
report = classification_report(y_test,predictions,output_dict=True)

print("Accuracy of the model: ",report['accuracy'])

Time taken to train the model: 1.0 min 15.770126581192017 sec
Accuracy of the model:  0.968047619047619


In [6]:
model = SVC(kernel='poly',degree=4) # kernel='poly',degree=2

start = time.time()
model.fit(x_train,y_train)
end = time.time()
print("Time taken to train the model: %s min %s sec"% ((end-start)//60,(end-start)%60))
predictions = model.predict(x_test)
report = classification_report(y_test,predictions,output_dict=True)

print("Accuracy of the model: ",report['accuracy'])

Time taken to train the model: 1.0 min 54.2788360118866 sec
Accuracy of the model:  0.9526190476190476


---
Time taken incase of polynomial kernel with D value equal to 2 is 1 min 15 sec.
Accuracy is 0.968

Time taken incase of polynomial kernel with D value equal to 4 is 1 min 54 sec.
Accuracy is 0.953

Incase of D value 2 the amount of time taken is less and the accuracy is more than that of D value 4 according to our observation.